lib/Sema/PreCheckExpr.cpp

```diff
@@ -238,6 +238,32 @@ static bool diagnoseRangeOperatorMisspell(DiagnosticEngine &Diags,
   return false;
 }
 
+static bool diagnoseNonexistentPowerOperator(DiagnosticEngine &Diags,
+                                        UnresolvedDeclRefExpr *UDRE,
+                                        DeclContext *DC) {
+  auto name = UDRE->getName().getBaseIdentifier();
+  if (!name.isOperator())
```

Comment:
You can use `if (!(name.isOperator() && name.is("**"))` condition to avoid nesting.